extension/basicauthextension/extension.go

```diff
@@ -177,3 +208,28 @@ func (a *authData) GetAttribute(name string) interface{} {
 func (*authData) GetAttributeNames() []string {
 	return []string{"username", "raw"}
 }
+
+type basicAuthRoundTripper struct {
+	base     http.RoundTripper
+	authData *ClientAuthSettings
+}
+
+func (b *basicAuthRoundTripper) RoundTrip(request *http.Request) (*http.Response, error) {
+	newRequest := request.Clone(request.Context())
+	newRequest.SetBasicAuth(b.authData.Username, b.authData.Password)
+	return b.base.RoundTrip(newRequest)
+}
+
+func (ba *basicAuth) RoundTripper(base http.RoundTripper) (http.RoundTripper, error) {
+	if strings.Contains(ba.clientAuth.Username, ":") {
```

Comment:
Is that the only thing that a username should not contain?

Reply:
I was referring to this-
https://en.wikipedia.org/wiki/Basic_access_authentication (See client side)

Reply:
I understand that, but is that the only char you have to guard against?

Reply:
Based on [this](https://tools.ietf.org/html/rfc2617#page-6) RFC2617 Specification, HTTP Basic Authentication username can contain any TEXT excluding the symbol `:`

Reply:
Thanks for the confirmation!